pkg/storage/segments/store.go

```diff
@@ -339,9 +344,9 @@ func (s *segmentStore) Repair(ctx context.Context, path storj.Path, lostPieces [
 	totalRepairCount := len(newNodes)
```

Comment:
While we at this, let's add some sanity checks to avoid hard to debug problems in future. Let's have a check that `totalNilNodes == len(newNodes)`. If the check fails, return a descriptive error.